File: Tests/Types/Output/self-referencing-type.ts
/* eslint-disable */
// tslint:disable

export class SelfReferencingType {
    public stringProperty: string;
    public selfProperty: SelfReferencingType;
    public selfList: SelfReferencingType[];
}

// outputid:605e91d7-ee13-4f1d-9b92-845bb3ace852

File: Tests/v10/SelfReferencingAnnotationsNullable/SelfReferencingType.cs
using KY.Generator;

namespace SelfReferencing;

[GenerateTypeScriptModel]
[GenerateStrict]
public class SelfReferencingType
{
    public string StringProperty { get; set; } = "";
    public SelfReferencingType? SelfProperty { get; set; }
    public List<SelfReferencingType> SelfList { get; set; } = [];
}
